dace/frontend/python/replacements.py

```diff
@@ -1582,3 +1592,1287 @@ def _matmult(visitor, sdfg: SDFG, state: SDFGState, op1: str, op2: str):
+                        true_state.add_edge(
+                            inp_node, None, tasklet,
+                            tasklet_params['inputs'][idx],
+                            dace.Memlet.from_array(inp_name, inp_data))
+                        idx += 1
+                for i, arg in enumerate(outputs):
+                    if isinstance(arg, str) and arg in sdfg.arrays.keys():
+                        out_name, _ = nested_sdfg_outputs[arg]
+                        out_data = nested_sdfg.arrays[out_name]
+                        out_node = true_state.add_write(out_name)
+                        true_state.add_edge(
+                            tasklet, tasklet_params['outputs'][i],
+                            out_node, None,
+                            dace.Memlet.from_array(out_name, out_data))
+
+                false_state = nested_sdfg.add_state(
+                    label=state.label + '_where_false')
+                nested_sdfg.add_edge(cond_state, false_state,
+                                     dace.InterstateEdge(cond_else))
+                nested_sdfg.add_edge(true_state, false_state,
+                                     dace.InterstateEdge())
+
+
+                codenode = state.add_nested_sdfg(
+                    nested_sdfg, sdfg,
+                    set([n for n, _ in nested_sdfg_inputs.values()]),
+                    set([n for n, _ in nested_sdfg_outputs.values()]))
+                me, mx = state.add_map(state.label + '_map', map_indices)
+                for arg in inputs + [where]:
+                    n = state.add_read(arg)
+                    conn, idx = nested_sdfg_inputs[arg]
+                    state.add_memlet_path(n, me, codenode,
+                                          memlet=dace.Memlet.simple(n, idx),
+                                          dst_conn=conn)
+                for arg in outputs:
+                    n = state.add_write(arg)
+                    conn, idx = nested_sdfg_outputs[arg]
+                    state.add_memlet_path(codenode, mx, n,
+                                          memlet=dace.Memlet.simple(n, idx),
+                                          src_conn=conn)
+                return
+
+        input_memlets = dict()
+        inp_conn_idx = 0
+        for arg, idx in zip(inputs, input_indices):
+            if isinstance(arg, str) and arg in sdfg.arrays.keys():
+                conn = tasklet_params['inputs'][inp_conn_idx]
+                input_memlets[conn] = Memlet.simple(arg, idx)
+                inp_conn_idx += 1
+        output_memlets = {
+            out_conn: Memlet.simple(arg, output_indices)
+            for arg, out_conn in zip(outputs, tasklet_params['outputs'])
+        }
+        state.add_mapped_tasklet(
+            tasklet_params['name'],
+            map_indices,
+            input_memlets,
+            tasklet_params['code'],
+            output_memlets,
+            external_edges=True
+        )
+
+
+@oprepo.replaces_ufunc('ufunc')
```

Comment:
Not necessary (is this an actual ufunc name?)

Reply:
No, all ufuncs will use this single method. I used a replacement operator, so I can have different functions for each one of the ufunc methods (currently only reduce, accumulate, and outer supported).